server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1777,25 +1831,34 @@ public void clusterStateProcessed(ClusterState oldState, ClusterState newState)
      * {@link SnapshotDeletionsInProgress.State#STARTED} or waiting entries in state {@link SnapshotDeletionsInProgress.State#WAITING}
      * that were moved to {@link SnapshotDeletionsInProgress.State#STARTED} in the returned updated cluster state.
      *
+     * @param projectId the project for repositories where deletions should be prepared. {@code null} means all projects
      * @param currentState current cluster state
      * @return tuple of an updated cluster state and currently executable snapshot delete operations
      */
-    private static Tuple<ClusterState, List<SnapshotDeletionsInProgress.Entry>> readyDeletions(ClusterState currentState) {
+    private static Tuple<ClusterState, List<SnapshotDeletionsInProgress.Entry>> readyDeletions(
+        ClusterState currentState,
+        @Nullable ProjectId projectId
+    ) {
```

Comment:
This is another example of different scopes for triggering next operation. In this case, it does not have the single repository scope, but either cluster wide or single project (when `projectId == null`).